TESTS/netsocket/dns/main.cpp

```diff
@@ -161,7 +162,18 @@ static void net_bringup()
     net = NetworkInterface::get_default_instance();
     nsapi_error_t err = net->connect();
     TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, err);
-    printf("MBED: IP address is '%s'\n", net->get_ip_address() ? net->get_ip_address() : "null");
+    const char *address = net->get_ip_address();
+
+    printf("MBED: IP address is '%s'\n", address ? address : "null");
+#if MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE == MESH
```

Comment:
Ah.. this is not working because value of `MESH` is not defined.

In case `MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE` is defined to be `MESH` then this comparison becomes

```
#if MESH == MESH
```

And because unspecified macros are replaced with zero, if is 
```
#if 0 == 0
```

And that is the problem, because
```
#if ETHERNET == MESH
```
is same as 
```
#if 0 == 0
```

So, you need to define some magic value for the `MESH`

```
#define MESH 1234
#if MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE == MESH
```
works.